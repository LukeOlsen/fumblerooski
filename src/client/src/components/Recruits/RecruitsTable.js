import React, { Component } from "react";

const RecruitsTable = props => {
  if (props.recruits[0].length > 0) {
    return (
      <div className="text-center">
        <table className="m-auto w-11/12">
          <thead>
            <tr>
              <th>Ranking</th>
              <th>Name</th>
              <th>Position</th>
              <th>Height</th>
              <th>Weight</th>
              <th>Stars</th>
              <th>Rating</th>
              <th>State</th>
            </tr>
          </thead>
          <tbody>
            {props.recruits[0] ? (
              props.recruits[0].map((recruit, id) => {
                return (
                  <tr key={id}>
                    <td className="border">{recruit.ranking}</td>
                    <td className="border">{recruit.name}</td>
                    <td className="border">{recruit.position}</td>
                    <td className="border">{recruit.height}</td>
                    <td className="border">{recruit.weight}</td>
                    <td className="border">{recruit.stars}</td>
                    <td className="border">{recruit.rating}</td>
                    <td className="border">{recruit.stateProvince}</td>
                  </tr>
                );
              })
            ) : (
              <tr></tr>
            )}
          </tbody>
        </table>
      </div>
    );
  } else {
    return <span></span>;
  }
};

export default RecruitsTable;
