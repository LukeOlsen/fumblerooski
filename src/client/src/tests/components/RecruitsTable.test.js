import React from "react";
import renderer from "react-test-renderer";
import RecruitsTable from "../../components/Recruits/RecruitsTable";

const testRecruit = [
  {
    city: "Test",
    committedTo: "School",
    country: null,
    createdAt: null,
    height: 73,
    id: 3699,
    name: "Test",
    position: "CB",
    ranking: 1,
    rating: 1,
    recruitType: "HighSchool",
    school: "Test",
    stars: 5,
    stateProvince: "FL",
    team_id: 0,
    updatedAt: null,
    weight: 182,
    year: 2019,
  },
];

describe("Recruiting Table", () => {
  it("Renders a table of recruits based on data from db", () => {
    const component = renderer.create(
      <RecruitsTable recruits={[testRecruit]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
