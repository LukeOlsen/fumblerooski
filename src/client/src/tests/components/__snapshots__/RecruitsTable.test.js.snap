// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recruiting Table Renders a table of recruits based on data from db 1`] = `
<div
  className="text-center pb-2"
>
  <table
    className="m-auto mb-2 w-11/12"
  >
    <thead>
      <tr>
        <th>
          Ranking
        </th>
        <th>
          Name
        </th>
        <th>
          Position
        </th>
        <th>
          Height
        </th>
        <th>
          Weight
        </th>
        <th>
          Stars
        </th>
        <th>
          State
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="border"
        >
          1
        </td>
        <td
          className="border"
        >
          Test
        </td>
        <td
          className="border"
        >
          CB
        </td>
        <td
          className="border"
        >
          73
        </td>
        <td
          className="border"
        >
          182
        </td>
        <td
          className="border"
        >
          5
        </td>
        <td
          className="border"
        >
          FL
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
